src/components/views/Registration.js

```diff
@@ -67,6 +68,40 @@ const Registration = () => {
     checkAvailability()
   }, [username]) // the function gets called whenever the username changes
 
+
+  const getGeoLocation = async (userId) => {
+
+    // how to handle if we get user location
+    const successCallback = (position) => {
+      const latitude = position.coords.latitude;
+      const longitude = position.coords.longitude;
+      setLocation([latitude, longitude]);
+    };
+
+    // how to handle if user denies access to location
+    const errorCallback = (error) => {
+      console.log(`Error: ${error.code}`);
+      console.log(`Error Message: ${error.message}`);
+      window.alert("Your location could not be determined. You were set to the default location (0\"N, 0\"E).");
+    };
+
+    // if geolocation is not supported by the browser, do so
+    if (!navigator.geolocation) {
+      window.alert("Geolocation is not supported by your browser. You were set to the default location (0\"N, 0\"E).");
+    } else {
+      navigator.geolocation.getCurrentPosition(successCallback, errorCallback);
+    }
+
+    console.log(location);
+    await api.post(`/users/${userId}/location`, {
+      latitude: location[0],
+      longitude: location[1]
+    });
+
+  }
```

Comment:
Firstly, I see that this works, but this solution is not extensible. Can you either pull this function out of the component / or, just use the api interceptor as discussed.

Reply:
E.g. we might decide to update the location when getting Black Cards, because localStorage keeps intact, so the user remains logged in even if shutting down the pc and flying to Bangkok.

Reply:
i like the second point very much. i'll see that i can put it in the interceptor. I think that way we have a good compromise between keeping the functionality simple and creating a geolocation watch that monitors all the time.

Reply:
> Firstly, I see that this works, but this solution is not extensible. Can you either pull this function out of the component / or, just use the api interceptor as discussed.

pulling it out of the component is tricky, because it relies on the state that for some reason could not be set within the function itself (got a react error when i tried)